forte/utils/payload_factory.py

```diff
@@ -0,0 +1,159 @@
+from abc import abstractmethod
+from multiprocessing.sharedctypes import Value
+from typing import Callable
+import numpy as np
+from forte.data.ontology.top import Meta
+
+
+class PayloadFactory:
+    """
+    A class that handles the creation of payloads. It can register meta data and check its validity.
+    """
+
+    def __init__(self):
+        self.valid_meta = (
+            {}
+        )  # map from meta data name to function that handles this meta data
+
+    def register(self, meta: Meta):
+        """
+        A function that registers a meta data type into the factory.
+
+        Args:
+            meta_name: a Generic object that is used to register a Payload meta data type.
+        """
+        if meta.source_type not in ("web", "local"):
+            raise ValueError("Meta data source must be either 'web' or 'local'")
+        self.valid_meta[type(meta)] = True
+
+    def check_meta(self, meta):
+        return type(meta) in self.valid_meta
+
+
+class Payloading:
+    """
+    An class that help mapping meta data to loading function.
+    """
+
+    def __init__(self):
+        self._factory = PayloadFactory()
+
+    def load_factory(self, factory):
+        self._factory = factory
+
+    @abstractmethod
+    def route(self, meta):
+        """
+        Convert the meta into a loading function that takes uri and read data
+        from the uri.
+
+        Args:
+            meta: Meta data
+
+        Returns:
+            a function that takes uri and read data from the uri.
+        """
+        pass
+
+
+class ImagePayloading(Payloading):
+    """
+    A class that helps mapping Image meta data to loading function.
+    """
+
+    def route(self, meta) -> Callable:
+        """
+        A function that takes and analyzes a meta data and returns a
+        corresponding loading function.
+
+        Args:
+            meta: a Meta object that is used to determine the loading function.
+
+        Raises:
+            ValueError: if meta is not a valid meta data.
+
+        Returns:
+            a function that takes uri and read data from the uri, and it returns the data.
+        """
+        if not self._factory.check_meta(meta):
+            raise ValueError(f"Meta data{meta} not supported")
+        if not meta.source_type in ("web", "local"):
+            raise ValueError("Meta data source must be either 'web' or 'local'")
+
+        if meta.source_type == "local":
+            try:
+                import matplotlib.pyplot as plt  # pylint: disable=import-outside-toplevel
```

Comment:
First, too many assumptions, such as source type, libraries that people can use.
Second, you can't do this just by hard-coding a few `if-else` branches, how are people going to provide their own loading methods?
Third, we won't rely on all these external library in Forte, it will make the library very hard to use. These parsing should be done out of the core library.

In addition, we introduce `import-outside-toplevel` once simply to reduce some legacy dependency. We don't want to use them anymore